photon-core/src/main/java/org/photonvision/vision/pipe/impl/AprilTagDetectionPipeParams.java

```diff
@@ -45,8 +50,27 @@ public boolean equals(Object obj) {
         if (getClass() != obj.getClass()) return false;
         AprilTagDetectionPipeParams other = (AprilTagDetectionPipeParams) obj;
         if (family != other.family) return false;
+
+        boolean sameQuadParams = false;
+        boolean sameDetectorParams = false;
         if (detectorParams == null) {
```

Comment:
When will these inputs be null? i'm having a hard time following the chain of null/equality checks is all.

Reply:
I don't know. I copied the existing logic figuring that it was there for a reason.

Reply:
For equals/hash code I usually accept my IDE auto code gen as truth. Don't put any stock in the old stuff.

Reply:
I didn't know that was a thing. I've updated them with the autogenerated ones.